docs/jax-101/02-jitting.ipynb

```diff
@@ -0,0 +1,570 @@
+    "\n",
+    "One way to deal with this problem is to rewrite the code to avoid conditionals on value. Another is to use special [control flow operators](https://jax.readthedocs.io/en/latest/jax.lax.html#control-flow-operators) like `jax.lax.cond`. However, sometimes that is impossible. In that case, you can consider jitting only part of the function. For example, if the most computationally expensive part of the function is inside the loop, we can jit just that inner part (though make sure to check the next section on caching to avoid shooting yourself in the foot):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "id": "OeR8hF-NHAML",
+    "outputId": "2b15ebbc-0fd8-4de1-e37f-a25fb5d4a522"
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "DeviceArray(30, dtype=int32)"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {
+      "tags": []
+     },
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# While loop conditioned on x and n with a jitted body.\n",
+    "\n",
+    "@jax.jit\n",
+    "def loop_body(prev_i):\n",
+    "  return prev_i + 1\n",
+    "\n",
+    "def g_inner_jitted(x, n):\n",
+    "  i = 0\n",
+    "  while i < n:\n",
+    "    i = loop_body(i)\n",
+    "  return x + i\n",
+    "\n",
+    "g_inner_jitted(10, 20)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "5XUT2acoHBz-"
+   },
+   "source": [
+    "If we really need to jit a function that has a condition on the value of an input, we can tell JAX to help itself to a less abstract tracer for a particular input by specifing `static_argnums`. The cost of this is that the resulting jaxpr is less flexible, so JAX will have to re-compile the function for every new value of the specified input. It is only a good strategy if the function is guaranteed to get limited different values."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {
+    "id": "2yQmQTDNAenY",
+    "outputId": "ab9737b2-b991-47fe-c3d7-ed4a39c38832"
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "10\n"
+     ]
+    }
+   ],
+   "source": [
+    "f_jit_correct = jax.jit(f, static_argnums=0)\n",
+    "print(f_jit_correct(10))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {
+    "id": "R4SXUEu-M-u1",
+    "outputId": "7b704cea-188e-438b-9497-a6369a147f0d"
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "30\n"
+     ]
+    }
+   ],
+   "source": [
+    "g_jit_correct = jax.jit(g, static_argnums=1)\n",
+    "print(g_jit_correct(10, 20))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "LczjIBt2X2Ms"
+   },
+   "source": [
+    "## When to use jit\n",
+    "\n",
+    "In many of the the examples above, jitting is not worth it:"
```

Comment:
Suggestion:

- Again, based on https://jax.readthedocs.io/en/latest/notebooks/thinking_in_jax.html#to-jit-or-not-to-jit

```suggestion
    "In many of the the examples above, JITting is not worth it:"
```